gnocchi/rest/api.py

```diff
@@ -525,6 +525,17 @@ def get_measures(self, start=None, stop=None, aggregation='mean',
                 },
             })
 
+        aggregations = []
+        for g in sorted(granularity, reverse=True):
+            agg = self.metric.archive_policy.get_aggregation(
+                aggregation, g)
+            if agg is None:
```

Comment:
this duplicates the check above?

Reply:
No, that only check granularity, not the method.

Reply:
```
         if aggregation not in self.metric.archive_policy.aggregation_methods:
             abort(404, {
                 "cause": "Aggregation method does not exist for this metric",
                 "detail": {
                     "metric": self.metric.id,
                     "aggregation_method": aggregation,
                 },
             })
```
this?

Reply:
Yes this checks the method; so we need both.

Not saying that it might be factorizable in a next patch